test/unit/ui/map.test.js

```diff
@@ -1485,6 +1485,25 @@ test('Map', (t) => {
     });
 
     t.test('#removeFeatureState', (t) => {
+
+        t.test('accepts "0" id', (t) => {
+            const map = createMap(t, {
```

Comment:
This test passes even without the code change. That happens because the checks in `removeFeatureState` fall to the final `else` clause and remove all the state for the source.

Reply:
👍 added nuance to the test to make sure 0 id's aren't interpreted as source-wide state deletion, by adding a second property that should stay put